### PR TITLE
made the Security Group backend throw the same error as AWS if the nb of sec groups limit is hit

### PR DESCRIPTION
**Pull request by costypetrisor**
I have used the default limits from:
- [https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html#limits_ec2](EC2-classic sec group limits)
- [https://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Appendix_Limits.html#vpc-limits-security-groups](EC-VPC sec group limits)

### CODE REVIEW

**Comment by costypetrisor** [comment]
I need to refactor this a bit. I'll let you know when it's ok

**Comment by costypetrisor** [comment]
Apologies for the long delay in regards to updating this PR :(

I did indeed had a an oversight: grants to other security groups do weigh in the overall limit of grants that are allowed for a security group.
I've fixed that, and extended the 2 unit tests I've added to cover this as well.

File: moto/ec2/models.py [hunk 1]
```diff
@@ -1415,6 +1429,19 @@ def authorize_security_group_egress(self,
                 if not is_valid_cidr(cidr):
                     raise InvalidCIDRSubnetError(cidr=cidr)
 
+        max_nb_rules = 50 if group.vpc_id else 100
```
Comment (spulec) [inline moto/ec2/models.py]: Can you generalize the logic between these two places a bit?

Comment (costypetrisor) [inline moto/ec2/models.py]: @spulec I've added a new commit. Have a look and let me know if you like the way I did it or not.
:)

**Comment by spulec** [comment]
Looks great, thanks!

**Pull request merged**